src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/GeneralApplicationCreationNotificationHandler.java

```diff
@@ -53,17 +61,26 @@ public List<CaseEvent> handledEvents() {
 
     private CallbackResponse notifyGeneralApplicationCreationRespondent(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
-        boolean isNotificationCriteriaSatisfied = isNotificationCriteriaSatisfied(caseData);
+
+        CaseData civilCaseData = caseDetailsConverter
+            .toCaseData(coreCaseDataService
+                            .getCase(Long.parseLong(caseData.getGeneralAppParentCaseLink().getCaseReference())));
+
+        CaseData updatedCaseData = solicitorEmailValidation.validateSolicitorEmail(civilCaseData, caseData);
```

Comment:
you can assign it to caseData instead of updatedCaseData and then you won't need further changes in the 'if' block

Reply:
we have to use updatedCaseData as variable used in lambda expression should be final or effectively final. Cheers!